cabal-install/src/Distribution/Client/IndexUtils.hs

```diff
@@ -276,11 +276,11 @@ getSourcePackagesAtIndexState verbosity repoCtxt mb_idxState mb_activeRepos = do
         IndexStateTime ts0 -> do
             when (isiMaxTime isi /= ts0) $
                 if ts0 > isiMaxTime isi
-                    then warn verbosity $
-                                   "Requested index-state " ++ prettyShow ts0
-                                ++ " is newer than '" ++ unRepoName rname ++ "'!"
-                                ++ " Falling back to older state ("
-                                ++ prettyShow (isiMaxTime isi) ++ ")."
+                    then die' verbosity $
+                                   "Stopping this command as the requested index-state=" ++ prettyShow ts0
+                                ++ " is newer than (" ++ prettyShow (isiMaxTime isi)
+                                ++ "), the most recent state of '" ++ unRepoName rname
+                                ++ "'. You could try 'cabal update' to bring down a later state or request an earlier timestamp for index-state."
```

Comment:
```suggestion
                                ++ "'. You could try 'cabal update' to bring down a later state or force the use of current local state with `--index-state=HEAD`."
```

you have to use a index-sate <= HEAD and it seems to me than is more frequent use HEAD than a older one